api/lds.proto

```diff
@@ -103,6 +103,13 @@ message Listener {
 
   // The type of draining to perform at a listener-wide level.
   DrainType drain_type = 8;
+
+  // A list of individual listener filters that make up the filter chain for sockets accepted with
+  // the listener. These filters are run before any in the 'filter_chains', and these filters have
+  // the opportunity to manipulate and augment the connection metadata that is used in connection
+  // filter chain matching. Order matters as the filters are processed sequentially right after a
+  // socket has been accepted by the listener, and before a connection is created.
+  repeated Filter listener_filter_chain = 9;
```

Comment:
There is a plural name -chains in this file. Please use plural name here as well. What is the difference between _filters vs _filter_chain? Can you please explain the semantics of "chain"?

Reply:
I think linking to the header file defining the filter interface would be helpful, since this "connection filter" is a new concept. We could also document it somewhere like https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/network_filters and link.

Reply:
This should be named `listener_filters`, since it's effectively a common part of `filters` from each filter chain.

Reply:
I think semantically, we have the concept of filter chains already for L4 and L7. A chain is just a pipeline of filters, specified in sequential order as a list. So, I think arguably this is another filter chain here.

Reply:
Currently, there is: `Listener::filter_chains` (list of `message FilterChain`) and `Listener::filter_chains::filters` (list of `message Filter`), which indeed is a chain of filters, but it's not named like that, because the name is already taken.

This PR adds `Listener::listener_filter_chain` (list of `message Filter`), which complements list in `Listener::filter_chains::filters` and not `Listener::filter_chains` , so I'm not sure how naming it `listener_filter_chain` can be a good idea.

Reply:
OK, yes, `listener_filters` makes more sense.

Reply:
Renamed. Will work on the docs once I get the Envoy implementation rebased.